src/quic/node_quic_buffer.cc

```diff
@@ -17,7 +17,7 @@ void QuicBufferChunk::MemoryInfo(MemoryTracker* tracker) const {
 
 size_t QuicBuffer::Push(uv_buf_t* bufs, size_t nbufs, DoneCB done) {
   size_t len = 0;
-  if (nbufs == 0 || bufs == nullptr || is_empty(bufs[0])) {
+  if (UNLIKELY(nbufs == 0 || bufs == nullptr || is_empty(bufs[0]))) {
```

Comment:
What if multiple buffers are provided but only the first one is empty? The last condition doesn’t account for that, right?

(Also, I think `bufs == nullptr` can be dropped here, it should already imply that `nbufs == 0` and the compiler would probably already be able to optimize it out, because `nbufs != 0 && bufs == nullptr` would be undefined behavior.)

Reply:
Yeah good point on the `is_empty()` check. To this point we've only ever pushed one in at a time so it hasn't been a problem.

Reply:
thanks for checking that, spotted a couple of other minor things in that function to fix up also :-)